### PR TITLE
Flip the range for fade-out

### PR DESCRIPTION
**Pull request by nithinp7**
Previously, fading in tiles went from 0.0 --> 1.0 and fading out tiles went from 1.0 --> 0.0. When these values are used to threshold a dither texture in a renderer, this small discrepancy might cause floating point disagreement between fading in tiles and fading out tiles. So instead, we now want fading in and fading out tiles to have the exact same dither threshold (fade percentage) and simply invert the resulting sample if it is fading out. Since the dither texture is sampled in screen-space, this allows the fading out tiles and fading in tiles to blend together perfectly without appearing transparent.

@kring came up with the idea and it works quite nicely!

### CODE REVIEW

File: Cesium3DTilesSelection/src/Tileset.cpp [hunk 1]
```diff
@@ -188,24 +189,25 @@ void Tileset::_updateLodTransitions(
...
 
       float currentPercentage =
           pRenderContent->getLodTransitionFadePercentage();
-      if (currentPercentage <= 0.0f) {
-        // Remove this tile from the fading out list if it is already at 0%.
+      if (currentPercentage >= 1.0f) {
+        // Remove this tile from the fading out list if it is already done.
         // The client will already have had a chance to stop rendering the tile
         // last frame.
+        pRenderContent->setLodTransitionFadePercentage(0.0f);
```
Comment (kring) [inline Cesium3DTilesSelection/src/Tileset.cpp:204]: So when the fade is complete, the fade percentage is reset to 0.0...

File: Cesium3DTilesSelection/src/Tileset.cpp [hunk 2]
```diff
@@ -1402,7 +1404,7 @@ void Tileset::_unloadCachedTiles() noexcept {
     const TileRenderContent* pRenderContent =
         pTile->getContent().getRenderContent();
     if (_options.enableLodTransitionPeriod && pRenderContent &&
-        pRenderContent->getLodTransitionFadePercentage() > 0.0f) {
+        pRenderContent->getLodTransitionFadePercentage() < 1.0f) {
```
Comment (kring) [inline Cesium3DTilesSelection/src/Tileset.cpp:1407]: ... but tiles with a fade less than 1.0 will not be unloaded.

So... tiles will pretty much never be unloaded, right?

**Review by kring — COMMENTED** [review]
This looks great except I think there's a memory leak. Memory usage goes up steadily even with the cache size at 1 byte, and it drops immediately when LOD transitions are disabled. See below for the likely cause.

**Comment by nithinp7** [comment]
@kring Thanks for spotting that, I just simplified the logic that keeps fading out tiles from being unloaded. Now it just checks if a tile is in the fading-out list (actually an unordered_set) before deciding to unload it. The logic seems cleaner, the memory leak is fixed, and the extra `std::unordered_set::find` call looks fast enough in the profiler. What do you think?

**Comment by kring** [comment]
That sounds good to me, thanks @nithinp7!

**Pull request merged**